lib/launcher.rb

```diff
@@ -119,18 +119,18 @@ def webrick_server(app)
     end
 
     def launch
-      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
-
-      http_app = http_app()
-      https_app = https_app()
-      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
-
       if SETTINGS.daemon
         check_pid
         Process.daemon
         write_pid
       end
 
+      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
+
+      http_app = http_app()
+      https_app = https_app()
+      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
```

Comment:
Could you keep this check before forking, if possible? I think the pre-reqs for HTTP/HTTPS apps could be checked trivially before forking occurs - it's just checking for presence of various settings. The more errors that are raised before forking, the easier it is for users to know about and debug failed startups.